source/common/tls/utility.cc

```diff
@@ -226,6 +226,36 @@ std::string Utility::generalNameAsString(const GENERAL_NAME* general_name) {
     }
     break;
   }
+  case GEN_OTHERNAME:
+    ASN1_TYPE* value = general_name->d.otherName->value;
+    if (value == nullptr) {
+      break;
+    }
+    switch (value->type) {
+    case V_ASN1_NULL:
+      break;
+    case V_ASN1_BOOLEAN:
+      san = value->value.boolean ? "true" : "false";
+      break;
+    case V_ASN1_ENUMERATED:
+      san = std::to_string(ASN1_ENUMERATED_get(value->value.enumerated));
+      break;
+    case V_ASN1_INTEGER:
+      san = std::to_string(ASN1_INTEGER_get(value->value.integer));
+      break;
+    case V_ASN1_OBJECT: {
+      char tmp_obj[256]; // OID Max length
+      if (OBJ_obj2txt(tmp_obj, 256, value->value.object, 1) < 0) {
+        break;
+      }
+      san.assign(tmp_obj);
+      break;
+    }
+    default:
+      str = value->value.asn1_string;
+      san.assign(reinterpret_cast<const char*>(ASN1_STRING_data(str)), ASN1_STRING_length(str));
+      break;
```

Comment:
There are a lot of other types https://github.com/google/boringssl/blob/dec5989b793c56ad4dd32173bd2d8595ca78b398/include/openssl/asn1.h#L123-L151

Is it safe treat them as string?

Reply:
it is also good to have test to cover those different types.

Reply:
> There are a lot of other types https://github.com/google/boringssl/blob/dec5989b793c56ad4dd32173bd2d8595ca78b398/include/openssl/asn1.h#L123-L151
> 
> Is it safe to treat them as string?

Emm... it seems they are all string in the end.
https://github.com/google/boringssl/blob/dec5989b793c56ad4dd32173bd2d8595ca78b398/include/openssl/base.h#L275-L290

Reply:
Added test cases for the different types. And yeah, it seems like the other types are all strings